cmd/lint.go

```diff
@@ -44,12 +43,28 @@ func lintCommandAction(cmd *cobra.Command, args []string) error {
 		return errors.Wrap(err, "locating package root failed")
 	}
 
-	ok, err := docs.IsReadmeUpToDate()
+	readmeFiles, err := docs.AreReadmesUpToDate()
 	if err != nil {
-		return errors.Wrapf(err, "can't check if %s file is up-to-date", docs.ReadmeFile)
+		return errors.Wrap(err, "checking readme files are up-to-date failed")
+	}
+
+	outdatedStr := ""
+	errStr := ""
+	for _, f := range readmeFiles {
+		if !f.UpToDate {
```

Comment:
I'm trying to capture all outdated file names here to display in the error message. It's not pretty again. Is there a better way to do so?

Reply:
In the other comment I suggested to return an error if something bad happens.

You can check if `docs.AreReadmesUpToDate()` returned an error and any `readmeFiles`. If so, you can print the error and conditionally iterate over `readmeFiles`. No need to concatenate single error message. You can return a generic error  and use `cmd.Printf` to iterate over problematic files. We used to mix `cmd.Printf` and `errrors.Wrap` for such cases, you can take a look at `cmd/build.go` and `internal/builder/packages.go`.